lib/active_merchant/billing/gateways/orbital.rb

```diff
@@ -478,6 +509,32 @@ def add_customer_address(xml, options)
         end
       end
 
+      # Payment can be done through either Credit Card or Electronic Check
+      def add_payment_source(xml, payment_source, options = {})
+        if payment_source.instance_of?(ActiveMerchant::Billing::Check)
+          add_echeck(xml, payment_source, options)
+        else
+          add_creditcard(xml, payment_source, options[:currency])
+        end
+      end
+
+      # Adds Electronic Check attributes
+      def add_echeck(xml, check, options = {})
+        xml.tag! :CardBrand, 'EC'
+        xml.tag! :CurrencyCode, currency_code(options[:currency])
+        xml.tag! :CurrencyExponent, currency_exponents(options[:currency])
+        unless check.nil?
+          raise "Invalid Check Received. Error: #{check.validate}" unless check.validate.empty?
+
+          xml.tag! :BCRtNum, options[:routing_number]
```

Comment:
`routing_number` is an attribute of the `check` payment method type, so we should expect `check.routing_number` to grab what we need here.

Reply:
The check object is specific to the US banking system, i.e. it only accepts a nine digit routing number. Some countries such as Canada use an eight digit routing number.

Reply:
Hmm, I'm not entirely sure what constraints we have internally for ACH payment methods. In Spreedly Core, it looks like we treat all routing numbers as IBAN (which is the largest character count), so I think we could expect all routing numbers to come from the `payment_method` via `routing_number` attribute. Maybe @jessiagee can confirm for me.

Reply:
Erik confirmed that we will pass the `routing_number` with the `payment_method`, so I think we only need to be concerned about grabbing `check.routing_number` here.